keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,1001 @@
+   node that advertises the extended resource in DRA `ResorceSlice` or extended
+   resources backed by device plugin. When using DRA devices, it creates a
+   special `ResourceClaim` for the pod with the allocation result recording
+   which devices were picked. More details on this special `ResourceClaim`
+   follow below.  When using extended resources advertised for a node by device
+   plugin, the existing resource tracking reserves them.
+1. Introduce a field `ExtendedResourceClaimStatus` to pod's `Status`, such that:
+    - the kubelet can find the special `ResourceClaim` while looking for claims to prepare
+    - the kubelet can pass the devices to containers in the pod with the extended
+   resource requests, based on the container/extended resource to device
+   request mapping in the `ExtendedResourceClaimStatus`.
+
+Some quick clarifications around the basic concepts: extended resource backed by
+device plugin, extended resource backed by DRA, and dynamic resource.
+
+* extended resource backed by device plugin uses pod's
+  spec.containers[].resources.requests to request for resources, it consumes the capacity
+  from node's status.capacity. It is of type (string, int64)
+* dynamic resource uses `ResourceClaim` to request resources, and
+  `ResourceSlice` to provide resource capacity. A pod asks for resources through
+  resource claim requests in pod's spec.resources.claims. Dynamic resource type
+  is described in resource slice, simply speaking, it is a list of devices, with
+  each device being described as structured parameters.
+* extended resource backend by DRA is a combination of the two above. It uses pods'
+  spec.containers[].resources.requests to request for resources, and uses
+  `ResourceSlice` to provide resource capacity. Hence, it is of type (string, int64)
+   on the consumption side, and list of devices with a common
+  `ExtendedResourceName` on the capacity side.
+
+With these additions in place, the DRA devices can be consumed by extended resource
+requests, or by DRA resouce claims. The scheduler has everything it needs to support
+the dynamic allocation of devices to requests made through extended resource and
+resource claims. No static partition of resources between extended resources and
+resource claims is needed. The kubelet and DRA driver has everything they need
+to admit a pod and pass the allocated devices to the containers in the pod to run.
+
+Note the following cluster setup configuration and constraint:
+
+* One node in cluster can have an extended resource backed by DRA, and another node in the
+cluster can have the same named extended resource backed by Device Plugin.
+
+* One node in a cluster cannot have both extended resource backed by DRA, and same
+  named extended resource backed by device plugin at the same time. This implies
+  that either the resource is advertised through `ResourceSlice`,
+  or `Node`'s status.capacity.
```

Comment:
In my opinion this is important enough to add to non-goals.

Reply:
BTW, is this going to be enforced somehow?

Reply:
added it to the Non-Goals. It is not enforced, but my prototype implementation just picks the device plugin, and ignores the DRA when both are present on one node at the same time.